src/Digdir.Domain.Dialogporten.Infrastructure/InfrastructureExtensions.cs

```diff
@@ -55,6 +55,14 @@ public static class InfrastructureExtensions
             // Framework
             .AddValidatorsFromAssembly(thisAssembly, ServiceLifetime.Transient, includeInternalTypes: true)
             .AddDistributedMemoryCache()
+            .AddStackExchangeRedisCache(options =>
+            {
+                var infrastructureSettings = infrastructureConfigurationSection.Get<InfrastructureSettings>()
+                    ?? throw new InvalidOperationException("Failed to get Redis connection string. Infrastructure settings must not be null.");
```

Comment:
muligens en bedre måte å hente ut connection string på? 🤔

Reply:
Du må bare sette connection string direkte også sjekke om den er null etter det. Men dette er en grei måte å gjøre det på dersom AddStackExchangeRedis ikke støtter en factory som tar inn service collection. Det beste hadde vært om du kunne bruke IOptions mtp config validering, reload av config uten nedetid etc. 

Men her overskriver du in memory provideren. Skulle ikke den ligge over redis?

Reply:
True true. Ja burde bruke AddDistributedMemoryCache conditionally om vi er i development, sant?

Reply:
Tror ideen var å ha begge? 
Først sjekkes in-memory, så Redis, så hente fra whatever remote source

Noe ala dette?
https://learn.microsoft.com/en-us/answers/questions/1377433/multiple-types-of-distributed-cache-in-asp-net-cor

Reply:
aha, makes sense